src/execution/compiler/operator/hash_join_translator.cpp

```diff
@@ -231,8 +275,17 @@ void HashJoinTranslator::CheckJoinPredicate(WorkContext *ctx, FunctionBuilder *f
       auto left_mark = codegen->AccessStructMember(codegen->MakeExpr(build_row_var_), build_mark_);
       function->Append(codegen->Assign(left_mark, codegen->ConstBool(false)));
     }
-    // Move along.
-    ctx->Push(function);
+
+    // var probeRow : ProbeRow
+    auto probe_row_type = codegen->MakeExpr(probe_row_type_);
+    auto probe_row = codegen->MakeExpr(probe_row_var_);
+    function->Append(codegen->DeclareVarNoInit(probe_row_var_, probe_row_type));
+    // Fill row.
+    FillProbeRow(ctx, function, codegen->MakeExpr(probe_row_var_));
+    // joinConsumer(queryState, pipelineState, buildRow, probeRow);
+    std::initializer_list<ast::Expr *> args{GetQueryStatePtr(), codegen->MakeExpr(GetPipeline()->GetPipelineStateVar()),
+                                            codegen->MakeExpr(build_row_var_), codegen->AddressOf(probe_row)};
+    function->Append(codegen->Call(join_consumer_, args));
```

Comment:
Only call the outer join if you have a left outer-join. Otherwise, inline the consumption code.